### PR TITLE
Resolve deprecation warnings for example-basic

### PR DESCRIPTION
**Pull request by MikeMcC399**
This PR resolves the deprecation warnings produced by running [.github/workflows/example-basic.yml](https://github.com/cypress-io/github-action/blob/master/.github/workflows/example-basic.yml). It solves one item of issue https://github.com/cypress-io/github-action/issues/648 "Deprecated Ubuntu 18.04 used in example workflows".

## Previous deprecation warnings

The previous deprecation warnings, for instance in run [3643305670](https://github.com/cypress-io/github-action/actions/runs/3643305670), were:

"Node.js 12 actions are deprecated. For more information see: https://github.blog/changelog/2022-09-22-github-actions-all-actions-will-begin-running-on-node16-instead-of-node12/. Please update the following actions to use Node.js 16: actions/checkout@v2"

and

"The ubuntu-18.04 environment is deprecated, consider switching to ubuntu-20.04(ubuntu-latest), or ubuntu-22.04 instead. For more details see https://github.com/actions/virtual-environments/issues/6002"

## Remaining warning

The remaining warning is:

"macOS-latest pipelines will use macOS-12 soon. For more details, see https://github.com/actions/runner-images/issues/6384"

There is nothing which needs to be done about this warning as the action is already successfully running macOS 12.6.1 through macOS-latest. It can be considered as a pure info message.

### CODE REVIEW

**Comment by MikeMcC399** [comment]
@jaffrepaul 

The result of the checks was successful, apart from the unrelated known problem with example-start / start-multiple-commas, so from my point of view this PR can be merged.

example-basic



## Detail

The PR runs resulted in:
108 successful, 1 skipped, and 1 failing checks

The skipped test was release. That is correct.
The failing test was example-start / start-multiple-commas. This is a known issue unrelated to this PR.

The relevant log for the workflow file [example-basic](https://github.com/cypress-io/github-action/actions/runs/3659480852/workflow) is https://github.com/cypress-io/github-action/actions/runs/3659480852 which was successful! 

The only remaining warning (which is really an info) is about macOS:

"macOS-latest pipelines will use macOS-12 soon. For more details, see https://github.com/actions/runner-images/issues/6384"

I went through each of the sub-logs and they each look correct. The operating system shown under "Set up job" is as expected everywhere.

**Comment by MikeMcC399** [comment]
- Setting to Draft status to defer to PR https://github.com/cypress-io/github-action/pull/658 which should be a higher priority. Once the other PR is merged I will re-enable this one.

**Comment by MikeMcC399** [comment]
The latest run [3706380258](https://github.com/cypress-io/github-action/actions/runs/3706380258) was successful, so this can be merged now.

**Review by jaffrepaul — APPROVED** [review]

**Review by jaffrepaul — APPROVED** [review]

**Pull request merged**
